### PR TITLE
Refinance

### PR DESCRIPTION
**Pull request by iri031**
Refinance connector. Refinancing among notional, aave v2, aave v3, compound, makerDAO .

- [ ] Tests
- [ ] Review and approve connector
- [ ] Deploy
- [ ] Enable
- [ ] Add to SDK
- [ ] Add to docs

### CODE REVIEW

File: contracts/mainnet/connectors/refinance-notional/helpers.sol [hunk 1]
```diff
@@ -0,0 +1,243 @@
+//SPDX-License-Identifier: MIT
+pragma solidity ^0.7.6;
+pragma experimental ABIEncoderV2;
+
+// import { Helpers } from "./helpers.sol";
+import { Basic } from "../../common/basic.sol";
+import { Token, NotionalInterface, BalanceAction, BalanceActionWithTrades, DepositActionType, AaveV2LendingPoolProviderInterface, AaveV2DataProviderInterface, AaveV2Interface, AaveV3PoolProviderInterface, AaveV3Interface, AaveV3DataProviderInterface } from "./interface.sol";
```
Comment (pradyuman-verma) [inline contracts/mainnet/connectors/refinance-notional/helpers.sol]:
```suggestion
import "./interface.sol";
```

File: contracts/mainnet/connectors/refinance-notional/helpers.sol [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+contract Helpers is Basic {
+	using SafeERC20 for IERC20;
+
+	enum Protocol {
+		AaveV2,
+		AaveV3,
+		Compound,
+		Notional
```
Comment (pradyuman-verma) [inline contracts/mainnet/connectors/refinance-notional/helpers.sol]: should be capital

File: contracts/mainnet/connectors/refinance-notional/helpers.sol [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+
+	function toUint16(uint256 value) internal pure returns (uint16) {
+		require(value <= type(uint16).max, "uint16 value overflow");
+		return uint16(value);
+	}
+
+	function toUint8(uint256 value) internal pure returns (uint8) {
+		require(value <= type(uint8).max, "uint8 value overflow");
+		return uint8(value);
+	}
```
Comment (pradyuman-verma) [inline contracts/mainnet/connectors/refinance-notional/helpers.sol]: add these in math.sol in common folder

File: contracts/mainnet/connectors/refinance-notional/helpers.sol (hunk 1 above)
Comment (iri031) [reply contracts/mainnet/connectors/refinance-notional/helpers.sol]: Done.

File: contracts/mainnet/connectors/refinance-notional/helpers.sol (hunk 2 above)
Comment (iri031) [reply contracts/mainnet/connectors/refinance-notional/helpers.sol]: Okay. Done.

File: contracts/mainnet/connectors/refinance-notional/helpers.sol (hunk 3 above)
Comment (iri031) [reply contracts/mainnet/connectors/refinance-notional/helpers.sol]: Done, also updated the [helpers.sol](https://github.com/Instadapp/dsa-connectors/pull/230/commits/a715112abc5f9da732f31da5301887f851bbf0ea#diff-4da74c75b5284c601e06c717df178e761cbb820aab11de406ef9d6437c5b52a0L74) of notional connector.

**Pull request closed without merging**